### PR TITLE
String.Capitalize() in C# now matches the behaviour of String::capitalize() in C++

### LINKED ISSUE #56941: `string.Capitalize` has different behavior in C#
### Godot version

4.0, 3.x

### System information

Win10

### Issue description

```csharp
"MyNode".Capitalize();
```
It returns `Mynode` in C#, but `My Node` (in GDScript) is expected.

### Steps to reproduce

above

### Minimal reproduction project

_No response_

### PR DESCRIPTION
**Pull request by IgorKordiukiewicz**
Fixes #56941

### CODE REVIEW

File: modules/mono/glue/GodotSharp/GodotSharp/Core/StringExtensions.cs [hunk 1]
```diff
@@ -284,6 +284,51 @@ public static string Capitalize(this string instance)
             return cap;
         }
 
+        private static string CamelcaseToUnderscore(this string instance, bool lowerCase)
+        {
+            string newString = string.Empty;
+            int startIndex = 0;
+
+            for(int i = 1; i < instance.Length; i++)
```
Comment (raulsntos) [inline modules/mono/glue/GodotSharp/GodotSharp/Core/StringExtensions.cs:292]: To follow our code style there should be a space before the parentheses in `for` and `if` statements. Your IDE can be configured to do this for you automatically on save (or manually).

**Review by raulsntos — COMMENTED** [review]
The code looks just about the same as the GDScript implementation, except for the missing `Trim()`. Also needs a few code style fixes.

However, I'm not sure if using internal calls would be preferred here since otherwise we need to make sure it's kept in sync.

**Comment by IgorKordiukiewicz** [comment]
Updated

**Review by raulsntos — APPROVED** [review]
I'm still not sure if using internal calls would be preferred, but the code is correct.

**Review by neikeq — APPROVED** [review]

**Comment by neikeq** [comment]
Thanks!

**Comment by akien-mga** [comment]
Cherry-picked for 3.5.

**Pull request merged**